.template/variants/web/app/javascript/build.js

```diff
@@ -0,0 +1,13 @@
+const watch = process.argv.slice(2).includes('--watch');
+
+require('esbuild').build({
+  entryPoints: ['app/javascript/application.js'],
+  inject: ['app/javascript/global.js'],
```

Comment:
I am not sure why this needs to be done in `inject` instead of being in the manifest file `application.js` ? 🤔  

I am also a bit concerned about the naming of `global.js` as what else should go in there? Why not just name it `'app/javascript/i18n.js'?

Reply:
Oh ok, forget about this, I should have include in comment or PR description. So there is an issue when I tried to make I18n-js works. I will try to demo step by step.

Without inject, when starting the rails app, there is an error involving with the I18n, say if I have this `application.ts`

```
import './vendor/';
import './translations/translations';

import './initializers/';
import './screens/';

console.log(I18n.t('hello'));
````
![Screen Shot 2565-01-27 at 15 41 41](https://user-images.githubusercontent.com/6965195/151322948-69f913f7-ceb9-4e8d-8246-4c04d1636c4d.png)

We can fix this issue by importing I18n where we need

```js
import I18n from 'i18n-js';
```

But still, there will be an error on the auto-generate translation file -> `/translations/translations.js` This file is auto-generated from i18n js task, so we can't modify it.

So I've been looking for the old webpack config and found that we actually had it defined on `ProvidePlugin` to automatically loaded it everywhere.

```js
const webpack = require('webpack');
      const plugins = [
        new webpack.ProvidePlugin({
          // Translations
          I18n: 'i18n-js',
        })
      ];
      
environment.config.set('plugins', plugins);
```

Then I use esbuild [inject](https://esbuild.github.io/api/#inject) to have this same behaviour. This will load the module globally to make `I18n` works in the translations. That's why I named it `global`, so we can inject more things we need there.

Let me recheck again with the asset pipeline, maybe we can make it available -> https://github.com/fnando/i18n-js/tree/v3.9.0#rails-app-with-asset-pipeline (I've tried this but not working. But I have a new idea, maybe we can modify `config/manifest.js` to load it 🤔 )

Reply:
Tried adding `require` on the sprocket's `app/assets/config/manifest.js` but not working. As I checked, It actually does not add it to compiled application.js. Now it all depends on the esbuild to compile. Seems the [app/assets/config/manifest.js](https://github.com/rails/sprockets/blob/master/UPGRADING.md#manifestjs) can only contains link.

Reply:
Thank you for all the details. We shall go with this approach then ✌️